tests/tests/compiler/futures/async_function_called_different_fail.leo

```diff
@@ -0,0 +1,44 @@
+/*
```

Comment:
Could we add another test where `finalize1` and `finalize2` both take a `u8` as a parameter? We should expect that test to pass.

Reply:
The difference would be that there are two async functions in the child program.